src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/Controls/Grid.cs

```diff
@@ -272,6 +272,7 @@ public bool ShowGridLines
         /// <summary>
         /// Returns a ColumnDefinitionCollection of column definitions.
         /// </summary>
+        [TypeConverter(typeof(ColumnDefinitionCollectionConverter))]
```

Comment:
Any reason why type converter is applied to `ColumnDefinitions` property rather than `class ColumnDefinitionCollection`?

Reply:
I would find that a higher risk, as it would affect uses outside of the Grid (and would get returned by TypeDescriptor). The main reason I would be reluctant to do so is because the converter does not really support converting the whole collection. It supports only quite a narrow albeit common case where properties like MinWidth, MaxWidth, Offset or SharedSizeGroup are not set.

Reply:
There is also a discussion that for that reason, the converter might only support conversion from string, not to string.

(And type converter applied to a property was an important distinguishing factor for the XAML parser to allow manipulating collections using attribute, though with public setter we have departed from that requirement.)